test/lexers/jsx-lexer.test.js

```diff
@@ -452,12 +452,20 @@ describe('JsxLexer', () => {
         done()
       })
 
+      it('unescapes key when i18nKey is not provided', (done) => {
+        const Lexer = new JsxLexer()
+        const content = '<Trans>I&apos;m Cielquan</Trans>'
+        assert.equal(Lexer.extract(content)[0].key, "I'm Cielquan")
+        done()
+      })
+
       it('supports the shouldUnescape options', (done) => {
```

Comment:
I find this a little counter intuitive and confusing. I would expect that `shouldUnescape` would unescape the `defaultValue`. If I understand correctly, that's not the case in `react-18next`, is that right?

Also, I'm wondering if we should add a test with a `i18nKey` property with escaped value. wdyt?

Reply:
It is a little confusing, but I'll try to explain as I understand it.

The `shouldUnescape` option means that after the translation is loaded/displayed, it will be unescaped. That's why when we see the `shouldUnescape` option, we **don't** unescape when extracting, so that it can be unescaped by react-i18next when it is loaded/displayed.

Also, I think it makes sense to add a test for a `i18nKey` with an escaped value, nice suggestion.